File: src/main/java/ru/isb/bot/utils/StringUtils.java
package ru.isb.bot.utils;

public class StringUtils {

    public static String replaceHTML(String text) {
        return text.replaceAll("<", "&lt;")
                .replaceAll(">", "&gt;")
                .replaceAll("!", "&#33;")
                .replaceFirst("```", "<pre>")
                .replaceFirst("```", "</pre>");
    }
}

File: src/main/java/ru/isb/bot/utils/StringUtils.kt
package ru.isb.bot.utils

class StringUtils {

    companion object {
        fun replaceHTML(text: String): String =
            text.replace("<".toRegex(), "&lt;")
                .replace(">".toRegex(), "&gt;")
                .replace("!".toRegex(), "&#33;")
                .replaceFirst("```".toRegex(), "<pre>")
                .replaceFirst("```".toRegex(), "</pre>")

    }
}